src/__tests__/Accordion.spec.js

```diff
@@ -1,37 +1,148 @@
+  const accordionItems = [
+    <AccordionItem data-testid="item-1" key="1">
+      <AccordionHeader targetId="1" data-testid="head-1">
+        Accordion Item 1
+      </AccordionHeader>
+      <AccordionBody accordionId="1" data-testid="body-1">
+        This is the body of first item.{' '}
+      </AccordionBody>
+    </AccordionItem>,
+    <AccordionItem data-testid="item-2" key="2">
+      <AccordionHeader targetId="2" data-testid="head-2">
+        Accordion Item 2
+      </AccordionHeader>
+      <AccordionBody accordionId="2" data-testid="body-2">
+        This is the body of second item.{' '}
+      </AccordionBody>
+    </AccordionItem>,
+    <AccordionItem data-testid="item-3" key="3">
+      <AccordionHeader targetId="3" data-testid="head-3">
+        Accordion Item 3
+      </AccordionHeader>
+      <AccordionBody accordionId="3" data-testid="body-3">
+        This is the body of third item.{' '}
+      </AccordionBody>
+    </AccordionItem>,
+  ];
+
+  it('should render a div with "accordion" class', () => {
+    const { getByTestId } = render(
+      <Accordion open="this accordion" toggle={() => {}} data-testid="test" />,
+    );
+    const node = getByTestId('test');
+    expect(node.tagName).toMatch(/div/i);
+    expect(node).toHaveClass('accordion');
+  });
+
+  it('should render flush prop', () => {
+    const { getByTestId } = render(
+      <Accordion
+        open="this accordion"
+        flush
+        toggle={() => {}}
+        data-testid="test"
+      />,
+    );
+    const node = getByTestId('test');
+    expect(node).toHaveClass('accordion');
+    expect(node).toHaveClass('accordion-flush');
+  });
+
+  it('should render custom tag', () => {
+    testForCustomTag(Accordion, {
+      open: 'this accordion',
+      flush: true,
+      toggle: () => {},
+    });
+  });
+
+  it('should render custom classes', () => {
+    testForCustomClass(Accordion, {
+      open: 'this accordion',
+      toggle: () => {},
+    });
+  });
+
+  it('should have second item showing and others collapsed', () => {
+    const { getByTestId } = render(
+      <Accordion open="2" toggle={() => {}} data-testid="accordion">
+        {accordionItems}
+      </Accordion>,
     );
 
-    expect(wrapper.find('.accordion').length).toBe(1);
+    expect(getByTestId('body-1')).not.toHaveClass('show');
+    expect(getByTestId('body-2')).toHaveClass('show');
+    expect(getByTestId('body-3')).not.toHaveClass('show');
   });
 
-  it('should render with "accordion-flush" class', () => {
-    const wrapper = mount(
-      <Accordion flush open="this accordion" toggle={() => {}} />,
+  it('should call toggle with clicked item id', () => {
+    const mockFn = jest.fn(() => {});
+    const { getByText } = render(
+      <Accordion open="1" toggle={mockFn} data-testid="accordion">
+        {accordionItems}
+      </Accordion>,
     );
 
-    expect(wrapper.find('.accordion-flush').length).toBe(1);
+    fireEvent.click(getByText(/accordion item 2/i));
```

Comment:
It would probably be best to use the `user-event` companion library instead of the raw events directly.